src/dragonball/src/api/v1/vmm_action.rs

```diff
@@ -488,7 +507,32 @@ impl VmmService {
         Ok(VmmData::Empty)
     }
 
+    /// Setup dragonball tracing.
+    fn setup_tracing(&self, trace_info: TraceInfo) -> VmmRequestResult {
+        let mut tracer = self.tracer.lock().unwrap();
+        tracer
+            .setup_tracing(trace_info)
+            .map(|_| VmmData::Empty)
+            .map_err(VmmActionError::TracingFailed)
+    }
+
+    /// Get dragonball tracer.
+    pub fn tracer(&self) -> Arc<Mutex<DragonballTracer>> {
+        self.tracer.clone()
+    }
+
+    /// end dragonball tracing.
+    #[instrument(skip(self))]
```

Comment:
Why all `self` is droped?
Is that because we don't have `Debug` trait for VmmService?

Reply:
I don't think we need to ouput the VmmService info to tracing.